tests/src/test_helpers/ssl_helpers.c

```diff
@@ -1161,6 +1278,12 @@ int mbedtls_test_ssl_build_transforms(mbedtls_ssl_transform *t_in,
     memset(key0, 0x1, keylen);
     memset(key1, 0x2, keylen);
 
+#if defined(MBEDTLS_CIPHER_C)
```

Comment:
Minor, partially pre-existing: I wonder if we could change that to `#if !defined(MBEDTLS_USE_PSA_CRYPTO)`, which still indicated `CIPHER_C` is available, the way the dependencies are set up. Actually, the check for `!USE_PSA_CRYPTO` is already here a few lines below and its `#endif` if next to this one's, so it would be a simple matter of merging the two.

The main advantage would show when we prepare 4.0 and make `USE_PSA_CRYPTO` no longer an option (always on), and remove code that becomes dead: here it would be clearer that the cipher-using code is dead.